clang/lib/CConv/CheckedRegions.cpp

```diff
@@ -28,18 +30,97 @@ using namespace clang;
 
 bool CheckedRegionAdder::VisitCompoundStmt(CompoundStmt *S) { 
   llvm::FoldingSetNodeID Id;
+  ast_type_traits::DynTypedNode DTN = ast_type_traits::DynTypedNode::create(*S);
+
   S->Profile(Id, *Context, true);
   switch (Map[Id]) {
-    case IS_UNCHECKED: return true;
-    case IS_CHECKED:   
-                        auto Loc = S->getBeginLoc();
-                        Writer.InsertTextBefore(Loc, "_Checked ");
-                        return false;
+    case IS_UNCHECKED:
+      if(isParentChecked(DTN) && !isFunctionBody(S)) {
+        auto Loc = S->getBeginLoc();
+        Writer.InsertTextBefore(Loc, "_Unchecked ");
+      }
+      break;
+    case IS_CHECKED:
+      if(!isParentChecked(DTN)) {
+        auto Loc = S->getBeginLoc();
+        Writer.InsertTextBefore(Loc, "_Checked ");
+      }
+      break;
+    default: llvm_unreachable("Bad flag in CheckedRegionAdder");
+  }
+
+  return true;
+}
+
+
+
+bool CheckedRegionAdder::VisitCallExpr(CallExpr *C) {
+  auto FD = C->getDirectCallee();
+  FoldingSetNodeID ID;
+  ast_type_traits::DynTypedNode DTN = ast_type_traits::DynTypedNode::create(*C);
+
+  C->Profile(ID, *Context, true);
+  if (FD && FD->isVariadic() &&
+      Map[ID] == IS_CONTAINED && isParentChecked(DTN)) {
+    auto Begin = C->getBeginLoc();
+    Writer.InsertTextBefore(Begin, "_Unchecked { ");
+    auto End = C->getEndLoc();
+    Writer.InsertTextAfterToken(End, "; }");
+  }
+
+  return true;
+}
+
+typedef std::pair<const CompoundStmt*, int> StmtPair;
+
+StmtPair
+CheckedRegionAdder::findParentCompound(const ast_type_traits::DynTypedNode &N) {
+  return findParentCompound(N, 1);
+}
+
```

Comment:
You could use an optional argument instead. That is, drop this code and add `= 1` after the declaration of `int distance`.

Reply:
And: add comment to say what this function is doing.

Reply:
Yeah that's nicer. Java got me.